### PR TITLE
Add unit test report to sonar analysis

### LINKED ISSUE #580: Set up unit tests count metric for TS on Next
Currently on Next (https://next.sonarqube.com/sonarqube/component_measures?id=sonarts&metric=tests) we have only java tests

### PR DESCRIPTION
**Pull request by vilchik-elena**
Fixes #580

### CODE REVIEW

File: sonarts-core/package.json [hunk 1]
```diff
@@ -59,6 +59,7 @@
     "awesome-typescript-loader": "3.3.0",
     "glob": "7.1.2",
     "jest": "21.2.1",
+    "jest-sonar-reporter": "^1.3.0",
```
Comment (stas-vilchik) [inline sonarts-core/package.json]: use exact version

File: sonarts-core/jest.config.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ module.exports = {
   },
   mapCoverage: true,
   moduleFileExtensions: ["js", "ts"],
+  testResultsProcessor: "jest-sonar-reporter",
```
Comment (stas-vilchik) [inline sonarts-core/jest.config.js]: [configure](https://github.com/3dmind/jest-sonar-reporter#customization) it to generate report in `coverage/`folder

Comment (stas-vilchik) [reply sonarts-core/jest.config.js]: or add `test-report.xml` to `.gitignore`, but I'd prefer the first option

Comment (vilchik-elena) [reply sonarts-core/jest.config.js]: @stas-vilchik for some reason it didn't work for "coverage" directory, may be dir was recreated for coverage

Comment (vilchik-elena) [reply sonarts-core/jest.config.js]: Added file to gitignore

**Pull request merged**